### DISCUSSION TITLE
Auto load on os start

### DISCUSSION THREAD

**Discussion by ahmadhasannudin** [sezanzeb/input-remapper#134] [Q&A]
how to auto load on os start? im using ubuntu

**Comment by sezanzeb** [comment]
should already happen. `sudo systemctl enable key-mapper`

**Reply by omgmajk** [reply]
I have to manually start the application in vmware kali everytime I log in to get the keyboard changes. Doesnt matter if I type in any of the following

>sudo systemctl enable key-mapper
or
>sudo systemctl start key-mapper

When I log on, I have to manually start the application GUI for it to make the changes I want.

**Reply by sezanzeb** [reply]
what does `systemctl status key-mapper` yield?

**Reply by omgmajk** [reply]
Never mind mate, I forgot to add my latest profile to "auto load" in the GUI. Thanks for a great program.

**Reply by gclintj** [reply]
Great and helpful program. Thank you!

As for this problem, still kinda confused.
Do I have to click "apply" and "Autoload" in a specific order? 
Sometimes it will work when ubuntu starts, but sometimes after I change the key mapping, even though it is applied and Autoload, it stops auto-load...

**Reply by sezanzeb** [reply]
you don't.

activate "autoload" to make it automatically apply when you log in

hit "apply" when you want to apply it right now

**Reply by kjozsa** [reply]
@omgmajk you saved my day, I was suffering with the same problem. Maybe this should have been obvious but I missed this config on the UI... 

Maybe README could get improved on this?

ps. thanks for this software, this at last allowed me to switch to Wayland.. having no mouse button mappings was a blocker for me :)

**Reply by lypanov** [reply]
Wasn't obvious to me either. Maybe rename it "Auto-apply"?

**Reply by sezanzeb** [reply]
> Maybe rename it "Auto-apply"?

Hm, I guess that makes sense. Anyone else having an opinion on this?

**Reply by Daniel-Chin** [reply]
Had the same issue of missing the "autoload" UI. But the existence of this thread kinda solves the issue via search engine. Also, GPT-5 is apparently aware of the "autoload" switch.

**Comment by blgram** [comment]
I have the same problem, it doesn't autoload at startup,  the command that is provided is not available in my device, key-mapper doesn't exist.

**Reply by sezanzeb** [reply]
use input-remapper instead of key-mapper

**Reply by blgram** [reply]
> use input-remapper instead of key-mapper

I have tried it before and since you mentioned I have tried again, but no luck. 
i get this error when i try input-remapper, 

`Failed to enable unit: Unit file input-remapper.service does not exist.`

**Reply by sezanzeb** [reply]
Then the installation probably wasn't successful. Try to install it again or try other installation methods. There should be `input-remapper.service` in `/usr/lib/systemd/system`

**Reply by blgram** [reply]
> Then the installation probably wasn't successful. Try to install it again or try other installation methods. There should be `input-remapper.service` in `/usr/lib/systemd/system`

I tried that, but I think I know the reason, first of all everytime I launch the software I have to put sudo password. 
I am still in look for the solution. I am new to Linux, so I am still exploring and understanding how things work in Linux.

**Reply by Tesla-Tank** [reply]
I had this problem when using the manual build. I switch to it since the fedora package isn't updating constantly, but someone managed to find a work around for this: `sudo systemctl enable input-remapper`
It worked for me after a restart.

**Reply by ValiantTurtle** [reply]
On Mint 22.1 I had to enable `input-remapper-daemon.service`. After that it seemed to recognize the alias and now responds to just input-remapper. So if your system does not respond to just input-remapper you might try this first:
`sudo systemctl enable input-remapper-daemon.service`

**Comment by makoConstruct** [comment]
I notice that [the arch package for this](https://aur.archlinux.org/cgit/aur.git/tree/input-remapper-git.install?h=input-remapper-git) (as confirmed in your readme) doesn't autostart. It did initially, but those lines in post_install are now commented out. Why? It seems to be a convention with arch not to autostart services, but that isn't in the guidelines, and it isn't argued for anywhere, and it seems somewhat insane when the service is the whole point of the package.

**Reply by sezanzeb** [reply]
@SammysHP what do you think?

**Reply by SammysHP** [reply]
Here's a quick summary of what units to enable automatically:

https://freedesktop.org/wiki/Software/systemd/Preset/

The consensus in Arch Linux is to keep units disabled by default and give the user the choice to enable them if wanted. Just installing a package should not start its programs automatically.

I'd also go as far as removing the whole install file. Updating the system doesn't automatically restart services (there are helper utilities to check which services use outdated libraries and need to be restarted). And I've never seen clean up of removed services either.

**Reply by makoConstruct** [reply]
> Updating the system doesn't automatically restart services (there are helper utilities to check which services use outdated libraries and need to be restarted). And I've never seen clean up of removed services either.

This just seems bad to me but you do you.

**Reply by SammysHP** [reply]
> This just seems bad to me but you do you.

Not at all! As a system administrator you want full control over the timing. Most programs can handle working with cached data for a while.

**Comment by ynaghibi** [comment]
I had this problem on arch and the solution is to move the mouse after login (especially when the KDE splash screen is visible) in order to make sure that the mouse is recognized. When you see the cursor stop moving for a moment, then this is a good sign that autoload is working in the background.
I hope we will see a fix for this issue, because I do not want to do this on every startup.
